plugins/pre-commit.sh

```diff
@@ -21,23 +21,22 @@
 # is used.
 
 # Find all staged Python files, and exit early if there aren't any.
-PYTHON_FILES=(`git diff --name-only --cached --diff-filter=AM | \
-  grep --color=never '.py$'`)
-if [ ! "$PYTHON_FILES" ]; then
+PYTHON_FILES=$(git diff --name-only --cached --diff-filter=AM | grep --color=never '.py$')
+if [ -z "${PYTHON_FILES}" ]; then
   exit 0
 fi
 
 ########## PIP VERSION #############
 # Verify that yapf is installed; if not, warn and exit.
-if [ -z $(which yapf) ]; then
+if ! command -v yapf >/dev/null; then
   echo 'yapf not on path; can not format. Please install yapf:'
   echo '    pip install yapf'
   exit 2
 fi
 ######### END PIP VERSION ##########
 
 ########## PIPENV VERSION ##########
-# if [ -z $(pipenv run which yapf) ]; then
+# if ! command -v pipenv run command -v yapf >/dev/null; then
```

Comment:
i don't think this works.  `command` is a shell built in, so trying to run it through `pipenv` will fail.  i think the fact this relies upon `pipenv` means also relying upon `which` is fine.  or you could try:

```bash
if ! pipenv run yapf --version 2>/dev/null 2>&1; then
```